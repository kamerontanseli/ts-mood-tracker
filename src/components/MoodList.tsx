import { reverse } from "ramda";
import * as React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { IMood } from "../store/reducers/moodHistory";
import { IAction, IState } from "../store/types";

export interface IMoodListProps {
  history: IMood[];
}

export const MoodList: React.SFC<IMoodListProps> = ({
  history
}: IMoodListProps) => (
  <div>
    <ul>
      {history.length < 1 ? (
        <li>No Moods recorded yet <Link to="/record">Record a mood</Link></li>
      ) : (
        <li><Link to="/record">Record a mood</Link></li>
      )}
      {history.map((mood, index) => (
        <li key={index}>
          {mood.value} - {new Date(mood.date).toDateString()}
        </li>
      ))}
    </ul>
  </div>
);

export const mapStateToProps = (state: IState): object => ({
  history: reverse(state.moodHistory.values)
});

export const mapDispatchToProps = (
  dispatch: (action: IAction) => void
): object => ({});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MoodList);
