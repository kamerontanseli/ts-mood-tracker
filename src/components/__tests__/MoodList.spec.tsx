import * as React from "react";
import * as renderer from "react-test-renderer";
import { MoodList, mapStateToProps, mapDispatchToProps } from "../MoodList";

jest.mock("react-router-dom");

describe('mapDispatchToProps', () => {
  it('should return an object', () => {
    expect(mapDispatchToProps(jest.fn())).toEqual({})
  })
})

describe('mapStateToProps', () => {
  it('should return the reverse of values', () => {
    const state = {
      moodHistory: {
        values: [{ date: 1, value: 1 }, { date: 2, value: 2 }, { date: 3, value: 3 }],
        nextMoodValue: 20
      }
    }
    expect(mapStateToProps(state)).toEqual({
      history: [{ date: 1, value: 1 }, { date: 2, value: 2 }, { date: 3, value: 3 }].reverse()
    })
  })
})

describe("MoodList", () => {
  it("should render", () => {
    const output = renderer.create(<MoodList history={[]} />);
    expect(output).toMatchSnapshot();
  });
  it("should render history", () => {
    const output = renderer.create(
      <MoodList
        history={[
          { date: 1542225226873, value: 20 },
          { date: 1542225226873, value: 30 }
        ]}
      />
    );
    expect(output).toMatchSnapshot();
  });
});
