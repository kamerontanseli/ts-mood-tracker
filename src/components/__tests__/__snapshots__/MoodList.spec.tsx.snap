// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodList should render 1`] = `
<div>
  <ul>
    <li>
      No Moods recorded yet 
    </li>
  </ul>
</div>
`;

exports[`MoodList should render history 1`] = `
<div>
  <ul>
    <li />
    <li>
      20
       - 
      Wed Nov 14 2018
    </li>
    <li>
      30
       - 
      Wed Nov 14 2018
    </li>
  </ul>
</div>
`;
